test/controllers/admin/deploy_groups_controller_test.rb

```diff
@@ -232,5 +232,57 @@
         refute @controller.instance_variable_get(:@preexisting_stages).empty?
       end
     end
+
+    describe "#merge_all_stages" do
+
+      describe "without a created stage" do
+        it "succeeds with no work to do" do
+          post :merge_all_stages, params: {id: deploy_group}
+          assert_redirected_to admin_deploy_group_path(deploy_group)
+        end
+      end
+
+      describe "with a create stage" do
+
+        let(:env) { environments(:staging) }
+        let(:deploy_group) { DeployGroup.create!(name: 'Pod 101', environment: env) }
+        let(:template_stage) { stages(:test_staging) }
+
+        let (:stage) {
+          post :create_all_stages, params: {id: deploy_group}
```

Comment:
prefer to crete state instead of calling a different action ... might have weird side-effects and makes debugging hard